### PR TITLE
[HDInsight] Support creating cluster with encryption at host

### PR DESCRIPTION
**Pull request by aim-for-better**
**Description**  

This PR adds a new parameter "encryption_at_host" to the cmdlet `az hdinsight create`.
It allows customer to create a cluster with encryption at host feature by running `az hdinsight create .. --encryption-at-host true`

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
HDInsight

File: src/azure-cli/azure/cli/command_modules/hdinsight/custom.py [hunk 1]
```diff
@@ -316,6 +316,12 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
         msi_resource_id=assign_identity
     )
 
+    if encryption_at_host:
+        if disk_encryption_properties:
+            disk_encryption_properties.encryption_at_host = encryption_at_host
+        else:
+            disk_encryption_properties = DiskEncryptionProperties(encryption_at_host=encryption_at_host)
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/hdinsight/custom.py:323]: I don't see the usage of `disk_encryption_properties`. Is it correct?

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/custom.py:323]: > I don't see the usage of `disk_encryption_properties`. Is it correct?

It is used at line 360.

**Review by MyronFanQiu — APPROVED** [review]

**Comment by MyronFanQiu** [comment]
@haroldrandom Could you help review this PR?

**Review by haroldrandom — APPROVED** [review]

**Pull request merged**